### ISSUE TITLE
Geometry_oM: Immutable list properties should be ReadOnlyCollection

### ISSUE THREAD

**Issue by IsakNaslundBh** [BHoM/BHoM#877]
#### Broken rules:

The PlanarSurface got made immutbale here https://github.com/BHoM/BHoM/pull/861
The internalBoundaries is still a list though, should instead be made a `ReadOnlyCollection<ICurve>` instead.

**Comment by kThorsager** [comment]
Yes, will need some extra action in `BH.Engine.Geometry.Query.InternalEdges` to return a List but seems like a contained problem at least

**Comment by kThorsager** [comment]
I retract my statement as there naturally is a use of the internal Boundaries in Revit_Toolkit.

**Comment by kThorsager** [comment]
To be resolved after https://github.com/BHoM/Revit_Toolkit/issues/722
